app/views/comments/_commentable.html.erb

```diff
@@ -62,16 +62,18 @@
     <% end %>
   </ul>
 
-  <div id="kudos_message"></div>
-
-  <h3 class="landmark heading"><a name="comments"><%= ts("Comments") %></a></h3>
 
+  <div id="kudos_message"></div>
+  <!-- note mixed responsibilities: kudos and comment messages -->
   <%= flash_div :comment_error, :comment_notice %>
 
   <% if @work && (@chapter.blank? || @chapter.posted?) %>
     <%= render "kudos/kudos", :kudos => @kudos, :guest_kudos_count => @work.guest_kudos_count, :commentable => @work %>
   <% end %>
 
+
+  <h3 class="landmark heading"><a id="comments"><%= ts("Comments") %></a></h3>
```

Comment:
Thanks! Will do!

To extract `kudos_error` and `kudos_notice` from comments `flash_div` was also my original thought. But then it looked like `comment_error` and `comment_notice` are used in many places, including some css. I had an impression that everything must be divided, but maybe I am wrong and it's easier than I thought.

Anyway, it seemed like this division should be its own task, as I thought moving just comments div shouldn't really have an impact on it.
But I haven't considered the use case you presented (thanks!).

So definitely agree with you and will try to go with your suggestion. It will take me a while though to dig into `flash_div` notice/errors matters, hence no changes are coming anytime soon.

Reply:
You did it exactly the right way! I gave it a quick check on my dev environment in addition to reading through the code and I don't see any new problems.